src/shogun/classifier/LDA.cpp

```diff
@@ -66,18 +68,34 @@ bool CLDA::train_machine(CFeatures *data)
 		if(!data->has_property(FP_DOT))
 			SG_ERROR("Specified features are not of type CDotFeatures\n")
 		set_features((CDotFeatures*) data);
+	} else {
+		data = get_features();
+		REQUIRE(data, "Features have not been provided!  Please pass in training features into train_machine() or into the constructor.\n")
```

Comment:
Please dont mention method names or technical terms such as "constructor" in error messages for users.
Just say "Features have not been provided.\n"